Moved get phonetic data out of the default function.

cmckinney-59 opened the pull request:
Moved processBaybayinText into its own file.

cmckinney-59 committed:
Moved get phonetic data out of the default function.
Moved processBaybayinText into its own file.

Jdsmith435 commented on src/components/Transliterator/Transliterator.jsx:
```diff
@@ -3,16 +3,28 @@ import './Transliterator.css'
+  if (text.includes("ch")) return PHONETIC_LETTERS.CH;
+  if (text.includes("c") && !text.includes("ch")) return PHONETIC_LETTERS.C;
+  if (text.includes("qu")) return PHONETIC_LETTERS.QU;
+  if (text.includes("j")) return PHONETIC_LETTERS.J;
```
Make the last if check an else check (default) or return PHONETIC_LETTERS.None instead of null

cmckinney-59 replied:
Fixed

Jdsmith435 commented on src/components/Utils/BaybayinTextProcessor.js:
```diff
@@ -0,0 +1,108 @@
+    return processedText;
+  }
+
+  function toLowerCaseText(text) {
```
Can you just call the default .toLowerCase()

cmckinney-59 replied:
Yes, I can. This was an artifact of when I built this in angular. Each one of these methods were in their own components but since we are just including it in one file it makes sense to not break up these one liners into their own methods because they are already self explanatory as it is.

cmckinney-59 replied:
Okay so in my next commit I removed all of the one line helper methods and inserted the logic directly into the processBaybayinText method. The logic is easy enough to understand that way. I only made one exception for 'removeAAfterConsonant' because the logic isn't as straight forward and I think for readability it would be better to have a named method.

Jdsmith435 commented on src/components/Utils/BaybayinTextProcessor.js:
```diff
@@ -0,0 +1,108 @@
+
+  function capitalizeSubsequentVowels(text) {
+    const vowelRegex = /([aeiou])([aeiou]+)/gi;
+    return text.replace(vowelRegex, (firstVowel, subsequentVowels) => {
```
if it is single line you can write it like this. You only need brackets + return if it is multiline 
    return text.replace(vowelRegex, (firstVowel, subsequentVowels) =>  firstVowel + subsequentVowels.toUpperCase());

cmckinney-59 replied:
Fixed

Jdsmith435 commented on src/components/Utils/BaybayinTextProcessor.js:
```diff
@@ -0,0 +1,108 @@
+  }
+
+  function removeDuplicateConsonants(text) {
+    const consonantRegex = /([bcdfghjklmnpqrstvwxyz])\1+/gi;
```
I know .net regexes are expensive to make but fast to run. You might want to consider making these global so the runtime only needs to create these once.

cmckinney-59 replied:
This is something that I may need to research how to do. I made a file named 'Roadmap' in the src folder containing this todo.

Jdsmith435 commented on src/components/Utils/BaybayinTextProcessor.js:
```diff
@@ -0,0 +1,108 @@
+  }
+
+  function addPlusIfConsonant(text) {
+    const consonantRegex = /[bcdfghjklmnpqrstvwxyz]/i;
```
What if you made a change to another consonantRegex but forgot to make the same change to another?

cmckinney-59 replied:
ahhh I see. This goes along with the global regex. I see why that is needed I will work on this in subsequent PRs.

Jdsmith435 commented on src/components/Utils/BaybayinTextProcessor.js:
```diff
@@ -0,0 +1,108 @@
+
+  function capitalizeVowel(text) {
+    return text.replace(/([.!?])\s*([aeiou])|(^|\s)([aeiou])/gi, (p1, p2, p3, p4) => {
+      return (p1 ? p1 : p3) + (p2 ? p2.toUpperCase() : p4 ? p4.toUpperCase() : '');
```
Single executable line, can just use () => x (single line) instead

cmckinney-59 replied:
fixed

Jdsmith435 commented on src/components/Utils/BaybayinTextProcessor.js:
```diff
@@ -0,0 +1,108 @@
+            transformedWord += '+';
+          }
+
+          if (i < word.length - 1 && consonantRegex.test(word[i]) && punctuationRegex.test(word[i + 1])) {
```
Can you combine the if on line 80 and on line 84 into the same logic? 
i < word.length - 1 && consonantRegex.test(word[i]) && ( punctuationRegex.test(word[i + 1]) || consonantRegex.test(word[i + 1]))

cmckinney-59 replied:
Done

Jdsmith435 reviewed (approved):
Im going to approve because I dont want to ruin your streak but the comments are still there for consideration

cmckinney-59 committed:
Broke down question dialog into smaller components and added a proper noun question component to dialog

cmckinney-59 committed:
Removed helper methods for all one line methods in BaybayinTextProcessor.

Added roadmap file

updated capitalizeSubsequentVowels and capitalizeVowel to be more inline

Combined 2 if statements  in addPlusIfConsonant

cmckinney-59 committed:
removed some sqiggly brackets that were breaking the code

cmckinney-59 committed:
Moved the PAGES content to the page content component

cmckinney-59 committed:
made a component inline

Jdsmith435 commented on src/components/Dialog/ProperNounQuestion.jsx:
```diff
@@ -0,0 +1,63 @@
+
+  if (!isProperNoun && !notPhonetic) {
+    showProperNounQuestion = (
+        <div>
+            <p>Is {enteredText} a proper noun?</p>
+            <div>
+                <button onClick={handleIsProperNoun}>Yes</button>
+                <button onClick={next}>No</button>
+            </div>
+        </div>
```
This is similar to PhoneticCharactersQuestion. Is a reusable component/props possible? There's a React Render pattern that might be helpful.

Jdsmith435 commented on src/components/Transliterator/Transliterator.jsx:
```diff
@@ -30,18 +43,16 @@ export default function Transliterator({
-      
+
       if (wordIncludesQuestion || wordIncludesCapital) {
         setDialogWord(key);
```
The state changes in this for loop are batched
